prow/jobs/test-infra/buildpack.yaml

```diff
@@ -300,8 +300,8 @@ prow_clonerefs_job_template: &prow_clonerefs_job_template
           memory: 1.5Gi
           cpu: 0.8
 
-kyma_cleaners_job_template: &prow_tools_job_template
-  run_if_changed: "^development/tools/"
+prow_tools_job_template: &prow_tools_job_template
+  run_if_changed: "^development/tools/(.*.go|.*.toml|.*.sh|go.mod|Dockerfile|Makefile)$
```

Comment:
the regex was ok - https://regex101.com/r/BO3jI9/1
you just need to use single quotes